packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -33,20 +61,40 @@ const LocationSearchControl = ({
   return null;
 };
 
-const LocationSearchStandalone = (props: LocationSearchProps) => {
-  const hasMounted = useRef(false);
+const LocationSearchStandalone = ({
+  onLoading = noop,
+  onResult = noop,
+  onResults = noop,
+  onClear = noop,
+  onError = noop,
+  ...props
+}: LocationSearchProps) => {
+  const geocoderRef = useRef(null);
 
   useEffect(() => {
-    if (!hasMounted.current) {
-      (createAmplifyGeocoder(props) as unknown as AmplifyLocationSearch).addTo(
-        `#${LOCATION_SEARCH_CONTAINER}`
-      );
+    const map = createAmplifyGeocoder(
+      props
+    ) as unknown as AmplifyLocationSearch;
+    geocoderRef.current = map;
 
-      hasMounted.current = true;
```

Comment:
Is there a reason we removed the `hasMounted` logic here?

Reply:
Was suggested [here](https://github.com/aws-amplify/amplify-ui/pull/2227#discussion_r911507362)

Reply:
That being said, think we would want to assign value of `geocoderRef` to the output of `createAmplifyGeocoder(props)` on ln 72 to prevent unnecessary re-assignments and calls to `createAmplifyGeocoder` on ln 75

Reply:
Could you elaborate on this please?

Reply:
@crnkyunicorn You can ignore that comment, think it is out of scope for this PR